api/envoy/config/filter/network/dubbo_proxy/v2alpha1/route.proto

```diff
@@ -0,0 +1,124 @@
+syntax = "proto3";
+
+package envoy.config.filter.network.dubbo_proxy.v2alpha1;
+option java_package = "io.envoyproxy.envoy.config.filter.network.thrift_proxy.v2alpha1";
+option java_multiple_files = true;
+option go_package = "v2";
+
+import "envoy/api/v2/route/route.proto";
+import "envoy/type/range.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "validate/validate.proto";
+import "gogoproto/gogo.proto";
+
+// [#protodoc-title: Dubbo route configuration]
+
+message RouteConfiguration {
+  // The name of the route configuration. Reserved for future use in asynchronous route discovery.
+  string name = 1;
+
+  // The interface name of the service.
+  string interface = 2;
+
+  // Which group does the interface belong to.
+  string group = 3;
+
+  // The version number of the interface.
+  string version = 4;
+
+  // The list of routes that will be matched, in order, against incoming requests. The first route
+  // that matches will be used.
+  repeated Route routes = 5 [(gogoproto.nullable) = false];
+}
+
+message Route {
+  // Route matching prarameters.
+  RouteMatch match = 1 [(validate.rules).message.required = true, (gogoproto.nullable) = false];
+
+  // Route request to some upstream cluster.
+  RouteAction route = 2 [(validate.rules).message.required = true, (gogoproto.nullable) = false];
+}
+
+message MethodMatch {
+  // The name of the method.
+  string name = 1;
+
+  message Parameter {
+    // Parameter index, starting from 0.
+    uint32 index = 1;
+
+    // Parameter types, only basic data types are supported.
+    string type = 2;
+
+    oneof parameter_match_specifier {
+      // If specified, header match will be performed based on the value of the header.
+      string exact_match = 3;
+
+      // If specified, header match will be performed based on range.
+      // The rule will match if the request header value is within this range.
+      // The entire request header value must represent an integer in base 10 notation: consisting
+      // of an optional plus or minus sign followed by a sequence of digits. The rule will not match
+      // if the header value does not represent an integer. Match will fail for empty values,
+      // floating point numbers or if only a subsequence of the header value is an integer.
+      //
+      // Examples:
+      //
+      // * For range [-10,0), route will match for header value -1, but not for 0,
+      // "somestring", 10.9,
+      //   "-1somestring"
+      envoy.type.Int64Range range_match = 4;
```

Comment:
If the type of the parameter is not an integer, is this range matching useful?

Reply:
Sorry, The type definition is not needed. I delete it.